src/internal/_curryN.js

```diff
@@ -0,0 +1,38 @@
+var arity = require('../arity');
+
+
+/**
+ * Internal curryN function.
+ *
+ * @private
+ * @category Function
+ * @param {Number} length The arity of the curried function.
+ * @return {array} An array of arguments received thus far.
+ * @param {Function} fn The function to curry.
+ */
+module.exports = function _curryN(length, recieved, fn) {
+  return function() {
+    var combined = [];
+    var argsIdx = 0;
+    var left = length;
+    var combinedIdx = 0;
+    while (combinedIdx < recieved.length || argsIdx < arguments.length) {
+      var result;
+      var rec = recieved[combinedIdx];
+      var recNotPlaceholder = rec == null || rec['@@functional/placeholder'] !== true;
+      if (combinedIdx < recieved.length &&
+          (recNotPlaceholder || argsIdx >= arguments.length)) {
+        result = rec;
```

Comment:
I think this would be clearer without `recNotPlaceholder`. Also, it seems wrong to access `recieved[combinedIdx]` before checking the bounds. I suggest the above become:

```javascript
if (combinedIdx < received.length &&
    (received[combinedIdx] == null ||
     received[combinedIdx]['@@functional/placeholder'] !== true ||
     argsIdx >= arguments.length)) {
  result = received[combinedIdx];
```